### PR TITLE
ft(ZENKO-3020): Add snapshot creation task

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by nicolas2bert — APPROVED** [review]

File: warpscript/utapi/createSnapshot.mc2 [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    { 'node' $nodeId } 'filterLabels' STORE
+
+    // Grab our passed timestamp and store it as the variable `endTimestamp`
+    $operation_info 'end' GET TOLONG 'endTimestamp' STORE
+
+    'utapi.snapshot.master' 'master_snapshot_class' STORE
+    'utapi.snapshot' 'snapshot_class' STORE
+    'utapi.checkpoint' 'checkpoint_class' STORE
+
+    $read_token $master_snapshot_class $filterLabels @utapi/fetchLatestRecord // Fetch lastest master snapshot
```
Comment (jonathan-gramain) [inline warpscript/utapi/createSnapshot.mc2]: ~lastest~ latest

File: warpscript/utapi/createSnapshot.mc2 [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    ] ->SET 'ignoredLabels' STORE
+
+    // Search for available snapshots
+    [
+      $read_token
+      $snapshot_class
+      $filterLabels
+    ] FINDSETS
+    DROP SWAP DROP // Keep only the label results
+    // Extract unquie labels
```
Comment (jonathan-gramain) [inline warpscript/utapi/createSnapshot.mc2]: ~unquie~ unique

**Review by dora-korpar — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

File: warpscript/utapi/createSnapshot.mc2 [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        'ops' GET 'createdOps' STORE
+        $checkpoint 'ops' GET 'checkpointOps' STORE
+
+        $createdOps // Push createdOps onto the stack for sumField
+        $checkpointOps KEYLIST
+        <%
+          'key' STORE
+          $key $checkpointOps $key GET @utapi/sumField
+        %> FOREACH
+        // Drop the remaining map returned by sumField form the stack
```
Comment (jonathan-gramain) [inline warpscript/utapi/createSnapshot.mc2]: ~form~ from

File: warpscript/util/getFromNested.mc2 [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  'desc'
+  <'
+Get a value from a nested map. returns null if either key does not exist.
+  '>
+  'sig' [ [ [ 'm"MAP' 'p:STRING' 'c:STRING' ] [ 'r:ANY' ] ] ] // Signature
+  'params' {
+    // Signature params description
+    'm' 'Nested map input'
+    'p' 'First level key'
+    'c' 'Decond level key'
```
Comment (jonathan-gramain) [inline warpscript/util/getFromNested.mc2]: ~Decond~ Second

**Review by jonathan-gramain — APPROVED** [review]
Only typos, did not check thoroughly the logic but it seems to do sensible stuff in general :wink:

**Pull request merged**